packages/arb-token-bridge-ui/src/components/TransactionHistory/TransactionsTableRowAction.tsx

```diff
@@ -76,13 +53,32 @@ export function TransactionsTableRowAction({
     return isClaiming || isClaimingCctp || !isConfirmed
   }, [isClaiming, isClaimingCctp, isConfirmed])
 
-  const isRedeemButtonDisabled = useMemo(
-    () =>
-      typeof isConnectedToArbitrum !== 'undefined'
-        ? !isConnectedToArbitrum
-        : true,
-    [isConnectedToArbitrum]
-  )
+  const isConnectedToCorrectNetworkForRedeem = useMemo(() => {
+    if (!chain) {
+      return false
+    }
+    return chain.id === tx.childChainId
+  }, [chain, tx.childChainId])
+
+  const handleRedeemRetryable = useCallback(async () => {
+    try {
+      if (!isConnectedToCorrectNetworkForRedeem) {
+        await switchNetworkAsync?.(tx.childChainId)
+      }
+      redeem()
```

Comment:
Shouldn't this be `await redeem()` or `return redeem()` so that it can be awaited later?

Reply:
redeem doesn't return anything so we can await it instead

Reply:
we don't need to await or return because the method doesn't return anything and nothing happens afterwards. All the logic is handled within redeem